src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,361 @@
+     * the dialog container. Clears the user input after processing.
+     */
+    private void handleUserInput() {
+        Label userText = new Label(userInput.getText());
+        Label dukeText = new Label(getResponse(userInput.getText()));
+        if(userInput.getText().equals("bye")){
+            dialogContainer.getChildren().addAll(
+                    DialogBox.getUserDialog(userText, new ImageView(userimg)),
+                    DialogBox.getDukeDialog(dukeText, new ImageView(dukeimg))
+            );
+            Platform.exit();
+            System.exit(0);
+            
+        }
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(userText, new ImageView(userimg)),
+                DialogBox.getDukeDialog(dukeText, new ImageView(dukeimg))
+        );
+        
+        
+        userInput.clear();
+    }
+    
+    private void dukeGreeting(){
+        Label userText = new Label(userInput.getText());
+        Label dukeText = new Label(getResponse(userInput.getText()));
+        dialogContainer.getChildren().addAll(
+                DialogBox.getDukeDialog(new Label(gui.dukeGreeting()), new ImageView(dukeimg))
+        );
+        userInput.clear();
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     * @return The string that dukebot outputs
+     */
+    String getResponse(String input) {
+        RequestType userRequest;
+        if(input.equals("bye")){
+            storage.writeToFile(tasks);
+            return gui.farewellMessage();
+        } else {
+            userRequest = duke.Parser.parse(input);
+            
+            switch (userRequest) {
+                case DEFAULT:
+                    return tasks.list();
+                case FIND:
+                    return tasks.find(input.substring(5));
+                case UNUSUAL:
+                    return gui.unusualRequest();
+                case DONE:
+                    return done(input);
+                case DELETE:
+                    return delete(input);
+                case DEADLINE:
+                    return deadline(input);
+                case EVENT:
+                    return event(input);
+                case TODO:
+                    return todo(input);
+            }
+        }
+        return "";
+    }
+    
+
+    /**
+     * Prints the following statement once an action is performed.
+     * 
+     * @param userInput Name of task
+     * @param actionType Type of action
+     */
+    public String echo(String userInput, String actionType) {
+        return gui.dukeResponse("Got it sir, I have " + actionType + " this task:\n "
+                + userInput + "\nNow you have " + tasks.getSize() + " tasks in the list.\n");
+
+    }
+    
+    private String done(String userInput) {
```

Comment:
Should the method name be a verb? I noticed this at many other places as well...